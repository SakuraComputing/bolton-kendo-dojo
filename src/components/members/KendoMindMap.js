import React, { Component } from 'react';

class KendoMindMap extends Component {

  render() {
    return (
      <div className="content-container mindmap-layout">
        <img src="/img/background/KendoMindMap.png" alt=""/>
      </div>
    )
  }
}
export default KendoMindMap;
