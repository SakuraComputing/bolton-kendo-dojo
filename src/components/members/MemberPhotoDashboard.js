import React from 'react'
import MemberPhotoFilter from './MembersPhotoFilter';
import MemberPhotos from './MembersPhotos';

const MemberPhotoDashboardPage = () => (
    <div>
        <MemberPhotoFilter />
        <MemberPhotos />        
    </div>

)
export default MemberPhotoDashboardPage;