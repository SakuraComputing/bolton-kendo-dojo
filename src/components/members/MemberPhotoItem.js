import React from 'react';

const MemberPhotoItem = (props) => {
    return (
        <div className="image-frame">
            <img className="image-item" src={`uploads/${props.filename}`} alt="photoU" />
            <div>
                {props.description}
            </div>
            <button onClick={props.onUploadDelete} className="button-small">Delete Photo</button>
        </div>
    );
}

export default MemberPhotoItem;
