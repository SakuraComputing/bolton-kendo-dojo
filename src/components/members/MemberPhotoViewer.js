import React from 'react'

const MemberPhotoViewer = () => {
    return (
        <div>
            <h1>Photo Viewer</h1>
        </div>
    )
}

export default MemberPhotoViewer
