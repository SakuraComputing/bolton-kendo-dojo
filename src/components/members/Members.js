import React from 'react';
import { Link } from 'react-router-dom';

const  Members = () => {
  return (
    <div className="members">
      <h1 className="members-header">Welcome to the Members Area</h1>
      <div className="members-area-container">
        <div className="members-area-link">
          <Link to="/memberphoto">Photos</Link>
          <p>Select the photographs and upload these with text for other members to see.</p>
        </div>
        <div className="members-area-link">
          <Link to="/mindmap">Kendo Mind Map</Link>
          <p>Enter the Kendo Mind Map section for all deeper kendo study</p>
        </div>
        <div className="members-area-link">
          <Link to="/memberphoto">Videos</Link>
          <p>Go into the video section and view private members videos</p>
        </div>
        <div className="members-area-link">
          <Link to="/members">Bulletin Board</Link>
          <p>Add a post to the bulletin board. Respond to other posts</p>
        </div>
      </div>
    </div>
  )
}
export default Members;
