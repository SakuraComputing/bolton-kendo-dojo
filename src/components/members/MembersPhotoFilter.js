import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setPhotoTextFilter, setPhotoDateSortOrder } from '../../actions/filterActions';

export class MembersPhotoFilter extends Component {

    onTextChange = (e) => {
        this.props.setPhotoTextFilter(e.target.value);
    }

    render() {
        return (
            <div>
                <div className="wrap-collapsible">
                        <input id="collapsible-search" className="toggle-search" type="checkbox"/>
                        <label htmlFor="collapsible-search" className="lbl-toggle-search">Search</label>
                        <div className="collapsible-content-search">
                            <div className="content-inner">
                                <form onSubmit={this.onFormSubmit} className="image-header">
                                    <div className="image-header-container">
                                        <label className="upload-label" htmlFor="filterText" >Search photo description</label>
                                        <input
                                            type="text"
                                            className="filter-text input-box"
                                            placeholder="Search By Description"
                                            value={this.props.filters.photoText}
                                            onChange={this.onTextChange}
                                            name="filterText"
                                        />                                        
                                    </div>
                                </form>
                            </div>
                        </div>
                    </div>

            </div>
        )
    }
}

const mapStateToProps = state => ({
    filters: state.filters
})

const mapDispatchToProps = (dispatch) => ({
    setPhotoTextFilter: (photoText) => dispatch(setPhotoTextFilter(photoText)),
    setPhotDateSortOrder: () => dispatch(setPhotoDateSortOrder())
})
export default connect(mapStateToProps, mapDispatchToProps)(MembersPhotoFilter);
