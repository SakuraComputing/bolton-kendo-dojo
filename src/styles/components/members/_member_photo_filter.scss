.filter-text {
    outline: none;
}

.filter-title {
    outline: none;
    margin-left: 10px;
}

.filter-dates {
    margin-left: 10px;
}