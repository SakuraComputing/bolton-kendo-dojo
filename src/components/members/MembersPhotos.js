import React from 'react';
import { connect } from 'react-redux';
import Spinner from '../../common/Spinner';
import { getMemberUploads, postMemberUploads, deleteMemberUploads } from '../../actions/uploadActions';
import MemberPhotoItem from '../../components/members/MemberPhotoItem';

export class MemberPhoto extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            file: null,
            description: '',
            errors: {}
        };
    };

    componentDidMount() {
        this.props.getMemberUploads();
    }

    onChange = (e) => {
        this.setState({ file: e.target.files[0] })  
    };

    onDescriptionChange = (e) => {
        this.setState({ description: e.target.value })
    }

    onFormSubmit = (e) => {
        e.preventDefault();

        const { file, description } = this.state; 

        let formData = new FormData();
        formData.append('image', file);
        formData.set('description', description);
        this.props.postMemberUploads(formData);
    }

    onUploadDelete = (index, e) => {
        const photoId = this.props.uploads.uploads[index]._id;
        this.props.deleteMemberUploads(photoId);
    }

    render() {

        let album;

        const { uploads, loading } = this.props.uploads;

        if(uploads === null || uploads === undefined || loading) {
            album = <Spinner />
        } else {
            if(uploads.length > 0) {
                album = uploads.map(((image, key) => {
                    return <MemberPhotoItem 
                            key={image._id}
                            filename={image.filename}
                            description={image.description}
                            onUploadDelete={this.onUploadDelete.bind(this, key)}
                        />
                }))
            } else {
                album = <h4>No uploads found....</h4>
            }
        }
        
        return (
            <div className="images-uploads content-container">
                <h1 className="image-title">Member Photograph Upload</h1>

                <div className="image-header">
                    <form onSubmit={this.onFormSubmit}>
                        <input type="file" className="upload-file" accept=".jpg,.jpeg,.png" name="image" id="file" onChange={this.onChange} />
                        <label className="upload-label" htmlFor="file">Upload Photo</label>
                        <input type="submit" value="Submit" className="button-small"/>
                        <label className="upload-label" htmlFor="description" >Add Photo Description</label>
                        <input className="upload-text input-box" type="text" name="description" onChange={this.onDescriptionChange}/>
                    </form>
                </div>
                
                <div className="image-container">
                    {album}
                </div>
            </div>
        )
    }
}
const mapStateToProps = (state) => ({
    errors: state.errors,
    uploads: state.uploads
})
export default connect(mapStateToProps, { getMemberUploads , postMemberUploads, deleteMemberUploads })(MemberPhoto);
