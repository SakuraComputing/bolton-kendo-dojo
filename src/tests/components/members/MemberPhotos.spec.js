import React from 'react'
import { shallow } from 'enzyme';
import { MemberPhoto } from '../../../components/members/MembersPhotos';
import uploads from '../../fixtures/uploads';

describe('Member Photos', () => {
    
    let photo, getMemberUploads, onUploadDelete,props;

    beforeEach(() => {
        getMemberUploads = jest.fn();
        onUploadDelete = jest.fn();
        props = { loading: false, uploads: uploads } ;
        photo = shallow(<MemberPhoto getMemberUploads={getMemberUploads} onUploadDelete={onUploadDelete} uploads={props} />)
    });

    it('should render correctly', () => {
        expect(photo).toMatchSnapshot();
    });

    it('should initialise present in `state`', () => {
        expect(photo.state()).toEqual({
            file: null,
            description: '',
            errors: {}
        });
    });
    
    describe('Form Handling', () => {
        describe('When typing in the description field', () => { 
            const description = 'New Phtoto';
    
            beforeEach(() => {
                photo.find('.upload-text').simulate('change', { target: { value: description }});
            })

            it('should accept an input on the description', () => {
                expect(photo.state().description).toEqual(description);
            });
        });
        
        describe('When selecting a file for input', () => {
            const file = null;
            beforeEach(() => {
                photo.find('.upload-file').simulate('change', { target: { files: file }});
            });
            it('should add the filename to the `state`', () => {    
                expect(photo.state().file).toEqual(file);
            });
        });

        describe('on update', () => {            

            let wrapper;
            let getMemberUploadsMock;
            let postMemberUploadsMock;

            beforeEach(() => {
                const onSubmitSpy = jest.fn();
                getMemberUploadsMock = jest.fn();
                postMemberUploadsMock = jest.fn();
                wrapper = shallow(
                <MemberPhoto 
                    uploads={uploads[0]} 
                    onSubmit={onSubmitSpy} 
                    getMemberUploads={getMemberUploadsMock}
                    postMemberUploads={postMemberUploadsMock}
                />)
            });
            it('should call onSubmit', () => {
                wrapper.find('form').simulate('submit', {
                   preventDefault: () => {}
                });
                expect(getMemberUploadsMock).toHaveBeenCalled();
            });
        });
        it('should display no uplaods when none are passed in', () => {
            getMemberUploads = jest.fn();
            onUploadDelete = jest.fn();

            props = { loading: false, uploads: [] } ;
            photo = shallow(<MemberPhoto getMemberUploads={getMemberUploads} onUploadDelete={onUploadDelete} uploads={props} />)
        });
    })
});