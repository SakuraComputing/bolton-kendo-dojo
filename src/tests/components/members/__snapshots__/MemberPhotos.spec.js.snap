// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Photos should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MemberPhoto
    getMemberUploads={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onUploadDelete={[MockFunction]}
    uploads={
      Object {
        "loading": false,
        "uploads": Array [
          Object {
            "_id": "5bfed7528032901512c0f933",
            "date": "2018-11-28T17:58:42.652Z",
            "description": "Kendo Photo Details",
            "filename": "image-1543168184113.png",
          },
          Object {
            "_id": "5bfed7528036661512c0f933",
            "date": "2018-11-21T17:58:42.652Z",
            "description": "More Details",
            "filename": "image-0000000000.png",
          },
          Object {
            "_id": "5bfed7528032901512c05rtd",
            "date": "2018-11-11T17:58:42.652Z",
            "description": "Extra Details",
            "filename": "image-3333333333.png",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="image-title"
        >
          Member Photograph Upload
        </h1>,
        <form
          className="image-header"
          onSubmit={[Function]}
        >
          <div>
            <label
              className="upload-file button-small"
              htmlFor="upload-file"
            >
              Get Photo
            </label>
            <input
              accept=".jpg,.jpeg,.png"
              id="upload-file"
              name="image"
              onChange={[Function]}
              type="file"
            />
          </div>
          <div>
            <label
              className="upload-label"
              htmlFor="submit"
            >
              Upload Photo
            </label>
            <input
              className="button-small"
              name="submit"
              type="submit"
              value="Submit"
            />
          </div>
          <div>
            <label
              className="upload-label"
              htmlFor="description"
            >
              Add Photo Description
            </label>
            <textarea
              className="upload-text input-box"
              col="2"
              name="description"
              onChange={[Function]}
              rows="2"
              type="text"
            />
          </div>
        </form>,
        <div
          className="image-container"
        >
          <MemberPhotoItem
            description="Kendo Photo Details"
            filename="image-1543168184113.png"
            onUploadDelete={[Function]}
          />
          <MemberPhotoItem
            description="More Details"
            filename="image-0000000000.png"
            onUploadDelete={[Function]}
          />
          <MemberPhotoItem
            description="Extra Details"
            filename="image-3333333333.png"
            onUploadDelete={[Function]}
          />
        </div>,
      ],
      "className": "images-uploads content-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Member Photograph Upload",
          "className": "image-title",
        },
        "ref": null,
        "rendered": "Member Photograph Upload",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <label
                className="upload-file button-small"
                htmlFor="upload-file"
              >
                Get Photo
              </label>
              <input
                accept=".jpg,.jpeg,.png"
                id="upload-file"
                name="image"
                onChange={[Function]}
                type="file"
              />
            </div>,
            <div>
              <label
                className="upload-label"
                htmlFor="submit"
              >
                Upload Photo
              </label>
              <input
                className="button-small"
                name="submit"
                type="submit"
                value="Submit"
              />
            </div>,
            <div>
              <label
                className="upload-label"
                htmlFor="description"
              >
                Add Photo Description
              </label>
              <textarea
                className="upload-text input-box"
                col="2"
                name="description"
                onChange={[Function]}
                rows="2"
                type="text"
              />
            </div>,
          ],
          "className": "image-header",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  className="upload-file button-small"
                  htmlFor="upload-file"
                >
                  Get Photo
                </label>,
                <input
                  accept=".jpg,.jpeg,.png"
                  id="upload-file"
                  name="image"
                  onChange={[Function]}
                  type="file"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Get Photo",
                  "className": "upload-file button-small",
                  "htmlFor": "upload-file",
                },
                "ref": null,
                "rendered": "Get Photo",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "accept": ".jpg,.jpeg,.png",
                  "id": "upload-file",
                  "name": "image",
                  "onChange": [Function],
                  "type": "file",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  className="upload-label"
                  htmlFor="submit"
                >
                  Upload Photo
                </label>,
                <input
                  className="button-small"
                  name="submit"
                  type="submit"
                  value="Submit"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Upload Photo",
                  "className": "upload-label",
                  "htmlFor": "submit",
                },
                "ref": null,
                "rendered": "Upload Photo",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "button-small",
                  "name": "submit",
                  "type": "submit",
                  "value": "Submit",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  className="upload-label"
                  htmlFor="description"
                >
                  Add Photo Description
                </label>,
                <textarea
                  className="upload-text input-box"
                  col="2"
                  name="description"
                  onChange={[Function]}
                  rows="2"
                  type="text"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add Photo Description",
                  "className": "upload-label",
                  "htmlFor": "description",
                },
                "ref": null,
                "rendered": "Add Photo Description",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "upload-text input-box",
                  "col": "2",
                  "name": "description",
                  "onChange": [Function],
                  "rows": "2",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <MemberPhotoItem
              description="Kendo Photo Details"
              filename="image-1543168184113.png"
              onUploadDelete={[Function]}
            />,
            <MemberPhotoItem
              description="More Details"
              filename="image-0000000000.png"
              onUploadDelete={[Function]}
            />,
            <MemberPhotoItem
              description="Extra Details"
              filename="image-3333333333.png"
              onUploadDelete={[Function]}
            />,
          ],
          "className": "image-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "5bfed7528032901512c0f933",
            "nodeType": "function",
            "props": Object {
              "description": "Kendo Photo Details",
              "filename": "image-1543168184113.png",
              "onUploadDelete": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "5bfed7528036661512c0f933",
            "nodeType": "function",
            "props": Object {
              "description": "More Details",
              "filename": "image-0000000000.png",
              "onUploadDelete": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "5bfed7528032901512c05rtd",
            "nodeType": "function",
            "props": Object {
              "description": "Extra Details",
              "filename": "image-3333333333.png",
              "onUploadDelete": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="image-title"
          >
            Member Photograph Upload
          </h1>,
          <form
            className="image-header"
            onSubmit={[Function]}
          >
            <div>
              <label
                className="upload-file button-small"
                htmlFor="upload-file"
              >
                Get Photo
              </label>
              <input
                accept=".jpg,.jpeg,.png"
                id="upload-file"
                name="image"
                onChange={[Function]}
                type="file"
              />
            </div>
            <div>
              <label
                className="upload-label"
                htmlFor="submit"
              >
                Upload Photo
              </label>
              <input
                className="button-small"
                name="submit"
                type="submit"
                value="Submit"
              />
            </div>
            <div>
              <label
                className="upload-label"
                htmlFor="description"
              >
                Add Photo Description
              </label>
              <textarea
                className="upload-text input-box"
                col="2"
                name="description"
                onChange={[Function]}
                rows="2"
                type="text"
              />
            </div>
          </form>,
          <div
            className="image-container"
          >
            <MemberPhotoItem
              description="Kendo Photo Details"
              filename="image-1543168184113.png"
              onUploadDelete={[Function]}
            />
            <MemberPhotoItem
              description="More Details"
              filename="image-0000000000.png"
              onUploadDelete={[Function]}
            />
            <MemberPhotoItem
              description="Extra Details"
              filename="image-3333333333.png"
              onUploadDelete={[Function]}
            />
          </div>,
        ],
        "className": "images-uploads content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Member Photograph Upload",
            "className": "image-title",
          },
          "ref": null,
          "rendered": "Member Photograph Upload",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <label
                  className="upload-file button-small"
                  htmlFor="upload-file"
                >
                  Get Photo
                </label>
                <input
                  accept=".jpg,.jpeg,.png"
                  id="upload-file"
                  name="image"
                  onChange={[Function]}
                  type="file"
                />
              </div>,
              <div>
                <label
                  className="upload-label"
                  htmlFor="submit"
                >
                  Upload Photo
                </label>
                <input
                  className="button-small"
                  name="submit"
                  type="submit"
                  value="Submit"
                />
              </div>,
              <div>
                <label
                  className="upload-label"
                  htmlFor="description"
                >
                  Add Photo Description
                </label>
                <textarea
                  className="upload-text input-box"
                  col="2"
                  name="description"
                  onChange={[Function]}
                  rows="2"
                  type="text"
                />
              </div>,
            ],
            "className": "image-header",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    className="upload-file button-small"
                    htmlFor="upload-file"
                  >
                    Get Photo
                  </label>,
                  <input
                    accept=".jpg,.jpeg,.png"
                    id="upload-file"
                    name="image"
                    onChange={[Function]}
                    type="file"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Get Photo",
                    "className": "upload-file button-small",
                    "htmlFor": "upload-file",
                  },
                  "ref": null,
                  "rendered": "Get Photo",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "accept": ".jpg,.jpeg,.png",
                    "id": "upload-file",
                    "name": "image",
                    "onChange": [Function],
                    "type": "file",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    className="upload-label"
                    htmlFor="submit"
                  >
                    Upload Photo
                  </label>,
                  <input
                    className="button-small"
                    name="submit"
                    type="submit"
                    value="Submit"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Upload Photo",
                    "className": "upload-label",
                    "htmlFor": "submit",
                  },
                  "ref": null,
                  "rendered": "Upload Photo",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "button-small",
                    "name": "submit",
                    "type": "submit",
                    "value": "Submit",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    className="upload-label"
                    htmlFor="description"
                  >
                    Add Photo Description
                  </label>,
                  <textarea
                    className="upload-text input-box"
                    col="2"
                    name="description"
                    onChange={[Function]}
                    rows="2"
                    type="text"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Add Photo Description",
                    "className": "upload-label",
                    "htmlFor": "description",
                  },
                  "ref": null,
                  "rendered": "Add Photo Description",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "upload-text input-box",
                    "col": "2",
                    "name": "description",
                    "onChange": [Function],
                    "rows": "2",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "textarea",
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <MemberPhotoItem
                description="Kendo Photo Details"
                filename="image-1543168184113.png"
                onUploadDelete={[Function]}
              />,
              <MemberPhotoItem
                description="More Details"
                filename="image-0000000000.png"
                onUploadDelete={[Function]}
              />,
              <MemberPhotoItem
                description="Extra Details"
                filename="image-3333333333.png"
                onUploadDelete={[Function]}
              />,
            ],
            "className": "image-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "5bfed7528032901512c0f933",
              "nodeType": "function",
              "props": Object {
                "description": "Kendo Photo Details",
                "filename": "image-1543168184113.png",
                "onUploadDelete": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "5bfed7528036661512c0f933",
              "nodeType": "function",
              "props": Object {
                "description": "More Details",
                "filename": "image-0000000000.png",
                "onUploadDelete": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "5bfed7528032901512c05rtd",
              "nodeType": "function",
              "props": Object {
                "description": "Extra Details",
                "filename": "image-3333333333.png",
                "onUploadDelete": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
